### ISSUE TITLE
Wayland Desktop Environment support

### ISSUE THREAD

**Issue by ragauskl** [bitgapp/Cursr#29]
Currently only X11 is officially supported, there are plans to add Wayland support in future but there's no specific date, maybe sometime late 2023 as there is a long list of higher priority tasks.

However if this issue get's a lot of interest (leave a thumbs up on issue are a comment) I will increase priority of adding support for Wayland and move it up the to-do list for Cursr

**Comment by jcrawfordor** [comment]
I'm seeing the same behavior as #51 on the Red Hat family, which now defaults to Wayland. Debian/Ubuntu has defaulted to Wayland on most hardware for a few years now, so the Linux version is currently non-working for a large portion of users who might try it out. Just adding this to inform your prioritization.

**Comment by ragauskl** [comment]
I started working on/playing around with Wayland support and there will be a lot of work required to provide it. There are higher priority tasks right now so I'll focus on them but continue working on/learning Wayland way of things on the side bit by bit.

It may take quiet a bit of time - but know that it's in progress. Can't provide a timeline atm, however currently I think I'll include priority updates with v1.8 and 1.9 and then may be able to fully focus back on Wayland (hopefully by that time knowing more about how to work with it). If not I'll provide further update at that time.

**Comment by TheInfamousAlk** [comment]
Hi, I'd like wayland support as well. Wayland has gotten to the point where it can handle modern gaming "out of the box" with no major tweaks. This is a relatively recent milestone in my opinion, and we will be seeing a lot more wayland users going forward. 

I just swapped linux distros and the new one (cachy os) uses wayland, and now I cannot use Cursr anymore. There really are no good alternatives out there either, so I'm just stuck either having to scale one of my monitors and make it look bad, or deal with a cursor that doesn't travel between monitors properly.

**Comment by ragauskl** [comment]
Yeah I have been doing some work on Arch Linux when I could find some time but now I got new setup and my main Linux is latest Ubuntu with Wayland as well. I'm well aware of this and it's high on priority list but so are couple other items that are way overdue, trying to catch up with the planned schedule now so hopefully soon there should be more regular updates again which will allow me to get to the Wayland task and dedicate full attention to it

**Comment by shermanhuman** [comment]
It's probably time to make this a priority.  The transition to Wayland is already the standard for most major "flagship" distributions. Here is the short list:

   * Ubuntu: Has been using Wayland by default for its main desktop since version 22.04 LTS.
   * Fedora: A long-time leader in the switch; it has defaulted to Wayland for years and is now actively removing the option for X11 sessions in its latest releases (Fedora 41+).
   * Pop!_OS: The new COSMIC desktop is built from the ground up to be Wayland-native and will be the default moving forward.
   * Debian: The standard GNOME version of Debian has defaulted to Wayland since Debian 10 (released in 2019).
   * KDE-based Distros: With the release of KDE Plasma 6, major distributions like Arch Linux, Fedora KDE, and KDE Neon have all switched their default session to Wayland.
   * Enterprise Linux (RHEL): Red Hat has officially deprecated X11 and is moving toward a Wayland-only environment for its next major release (RHEL 10).

The vast majority of new Linux desktop users are now landing in a Wayland environment by default.

**Comment by ragauskl** [comment]
As mentioned above I'm aware of how widely Wayland is used now and it is a high priority just not the top most one.

Believe me this doesn't bother anyone else more than it bothers me how much time this is taking and it's a responsibility always hanging above my head, but I am working alone on this (not planning to change this either) and managing some other things on the side which were reducing how much time I can spend on Cursr during last 2-3 years. I am trying to go back to working full time on this.

Wayland is currently planned to be released with v1.8, but I can't confirm yet if it will be with the first pre-release or a later one and when exactly it will be released, as there are things that do delay any work on the new version, such as new issues popping up (due to new complex user setups/new cases) and issues affecting core parts of Cursr are always a higher priority amongst some other things.

**Comment by shermanhuman** [comment]
Yeah, sorry.  My gentle nudge, on re-reading comes off as an entitled demand.  I get that you've got a complicated product with lot's of competing priorities.
